### PR TITLE
New Schedule for Minimum Airflow for VAV Boxes

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - This is a new feature / enhancement to VAV single and dual duct air terminals.
 - added a capability to turndown the minimum air flow below the design minimum using new turndown schedule field. The fraction multiplies the design minimum flow rate or the design minimum fraction.
 - this capability was added to the following single and dual duct VAV air terminals:  
   - AirTerminal:SingleDuct:VAV:Reheat
   - AirTerminal:SingleDuct:VAV:NoReheat
   - AirTerminal:SingleDuct:VAV:Reheat:VariableSpeedFan
   - AirTerminal:SingleDuct:VAV:HeatAndCool:Reheat
   - AirTerminal:SingleDuct:VAV:HeatAndCool:NoReheat
   - AirTerminal:DualDuct:VAV

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
  - [x] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [x] Check any new function arguments for performance impacts
 - [x] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (mjwitte) [inline design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:379]: "Operating" is a bit confusing here. Even though the other inputs are the "design" minimum, they also are used for operation, and the existing "Minimum Air Flow Fraction Schedule Name" is primarily operational. It's only used for sizing if the other min flow fields are blank.  How about "Minimum Air Flow Turndown Schedule Name" or "Minimum Air Flow Adjustment Schedule Name" or ???

Is this adjustment limited to zero to 1.0? Or would an adjustment >1.0 ever be applicable?

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (mjwitte) [inline design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:267]: It needs to be very clear that this schedule will be multiplied by the new schedule. That's a little confusing. I suppose few users would do that, they'd likely just use the new schedule with a fixed minimum here, but one could use both schedules together.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (mjwitte) [inline design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:421]: Feel free to rename this confusing struct `SysDesignParams` and its associated name `Sys`.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (mjwitte) [inline design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:477]: Same here, I've always found `DamperDesignParams` and `Damper` to be confusing.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (mjwitte) [inline design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:525]: My initial thought is to apply it. The user can always set this to 1.0 for the Winter and Summer design day types. But this needs to be explained clearly that it will get used. But I'd be ok with not applying it.  I guess it depends on the details of how this should be used relative to the referenced Standards.

**Review by mjwitte — COMMENTED** [review]
@Nigusse This looks good in general. Some comments.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (Nigusse) [reply design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:267]: I will try to add more detailed explanation.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (Nigusse) [reply design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:379]: I will use "Minimum Air Flow Turndown Schedule Name" as the new field name but will use "Adjustment" in the description.  I am thinking to limit to 0.0 to 1.0. I think users should use the design minimum fraction or the design minimum flow if they desire scale it up.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (Nigusse) [reply design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:421]: I will take the liberty to rename the struct when I get there. Thanks.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (Nigusse) [reply design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:477]: will do.

File: design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md
Comment (Nigusse) [reply design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:525]: I agree, applying it gives users the flexibility. If they choose not to do, they have the option to specify different schedule values for Winter and Summer design days. I will add note in the I/O Ref documentation and even in the idd if necessary.

Comment (Nigusse) [reply design/FY2020/NFP-Scheduling-Min-AirFlow-For-VAV-Boxes.md:525]: @mjwitte Thank you for the review and I will update the NFP based on your comments.

**Comment by Nigusse** [comment]
These variable renaming were suggested during NFP review. Diffs are not expected due to these commits.

**Comment by Nigusse** [comment]
@Myoldmopar @mjwitte I am not expecting any of the test suite files to fail for this branch. Is there any know issue in current develop?

**Comment by mjwitte** [comment]
@Nigusse There were some problems on Windows and some fixes over the weekend. Let's see how this comes back now that you've updated again.

**Comment by Nigusse** [comment]
@mjwitte I do not think the "x86_64-MacOS-10.13-clang" machine run failure are related to this branch. I just run a few selected files of "Failed Test" locally and all run successfully. All unit tests also run successfully. Am I missing something here?

**Comment by mjwitte** [comment]
@Nigusse I'm not suggesting the mac failures are related to this branch. Just getting @Myoldmopar 's attention to see if there's something up with the Mac CI machine or some real issue in develop for the Mac build.

**Comment by Myoldmopar** [comment]
Looking clean so far with these latest commits.  I'll take a look at the code changes as well.

File: src/EnergyPlus/DualDuct.hh
Comment (Myoldmopar) [inline src/EnergyPlus/DualDuct.hh:160]: The DamperNum argument is curious.  When converting functions into methods these usually vanish.  I'll have to see what's happening down further.

File: src/EnergyPlus/SingleDuct.cc
Comment (Myoldmopar) [inline src/EnergyPlus/SingleDuct.cc:293]: Yeah I'm not sure that the methods along with SysNum are being used _most_ effectively.  But I don't have time to dive deep enough to figure out the best path forward.

File: tst/EnergyPlus/unit/DualDuct.unit.cc
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/DualDuct.unit.cc:521]: :+1:

**Review by Myoldmopar — APPROVED** [review]
The implementation seems fine, and the unit test looks to provide at least some coverage of your changes.  The IDD and documentation are what I'd expect for this sort of change.

From a code change perspective, I like that the functions were made into methods, but unfortunately inside the methods, there was no refactor to actually use them as methods.  They still accept an integer component number argument and use it to lookup the instance many times in the methods.  I'm not going to require overhauling that now, but next time methods are made, we'll chat about best practices.

Thanks for these changes.

**Comment by Myoldmopar** [comment]
@mjwitte you seemed to feel pretty good about this earlier on in the development.  You have any lingering concerns?  If not then if CI is happy once complete this can merge in.

**Comment by mjwitte** [comment]
@Myoldmopar I took a quick look at just the IDD. That much looks good. I'll defer to you on reviewing the rest of this.

**Comment by Myoldmopar** [comment]
OK, yeah it'll merge as long as Windows doesn't see anything.

**Comment by Myoldmopar** [comment]
@Nigusse I tried to resolve the conflicts in this branch, but they are a little too involved.  Can you please pull develop in and resolve them and push back up?  Once complete and CI runs again this can go in.

**Comment by Nigusse** [comment]
OK, will do.

**Comment by Nigusse** [comment]
@Myoldmopar Resolved merge conflict and did cleanup.

**Comment by Nigusse** [comment]
@Myoldmopar working to resolve confclit and will push updated branch shortly after local test.

**Comment by Myoldmopar** [comment]
@Nigusse we've had a lot of activity today and it has resulted in some conflicts here.  I think they are primarily related to the uncontrolled terminal PR.  Can you please take a look at the conflicts, and if you have trouble, @mjwitte or I can help get things back in shape.  Thanks!

**Comment by Nigusse** [comment]
@Myoldmopar Sure, I am working on it. It will be done first thing in the morning.

**Comment by Nigusse** [comment]
@Myoldmopar @mjwitte Merged develop into the branch and resolved conflicts.

**Comment by Nigusse** [comment]
Run unit test suite locally and all passed.

**Comment by Nigusse** [comment]
@mjwitte I am seeing audit diffs that I was not expecting. Comparing this branch to develop I see the following four output variables that must have been removed from develop for "AirTerminal:SingleDuct:ConstantVolume:NoReheat" object:
- "Zone Air Terminal Sensible Heating Energy" 
- "Zone Air Terminal Sensible Cooling Energy" 
- "Zone Air Terminal Sensible Heating Rate"
- "Zone Air Terminal Sensible Cooling Rate"

Would you please confirm?

File: src/EnergyPlus/SingleDuct.cc
Comment (mjwitte) [inline src/EnergyPlus/SingleDuct.cc]: @Nigusse These were removed because they duplicate the same outputs that are already set up by the air distribution unit [here](https://github.com/NREL/EnergyPlus/blob/55f5f74c4f7b5b99e0110e5c7a158fd85bb20cc2/src/EnergyPlus/ZoneAirLoopEquipmentManager.cc#L470-L495).

Comment (Nigusse) [reply src/EnergyPlus/SingleDuct.cc]: @mjwitte Thanks. I will correct in this branch and push again.

**Review by mjwitte — APPROVED** [review]
@Nigusse I've pushed up a few minor text edits. I've run unit tests locally - all good. I've tested the new schedule with VAVSingleDuctReheat.idf - and it works.

There is no new example file for this. I'm not going to hold this up for that, but we need this somewhere in the testfiles. Perhaps there is an appropriate existing file to add this to (one that would correspond to applying ASHRAE Std 170).

Assuming CI comes back clean, I'll merge this soon.

**Comment by mjwitte** [comment]
@Nigusse Also - have you tested this with each of the affected terminal unit types?

**Comment by Nigusse** [comment]
@mjwitte  Yes, I have tested each combination. Also each combination has appropriate unit test. We can always add this new schedule field to selected existing test files.

**Pull request merged**
